### ISSUE TITLE
forceReadMVar empties its mvar

### ISSUE THREAD

**Issue by choener** [yamadapc/haskell-ascii-progress#17]
```
System/Console/AsciiProgress/Internal.hs
```

The line (184)
```
forceReadMVar mv v = tryTakeMVar mv >>= \m -> case m of
```
should be
```
forceReadMVar mv v = tryReadMVar mv >>= \m -> case m of
```
otherwise the contents will be replaced on every 2nd tick.

**Comment by yamadapc** [comment]
Mm... That's bad... Let's see if `tryReadMVar` is available on old versions of GHC.

https://travis-ci.org/yamadapc/haskell-ascii-progress/builds/96756009

**Comment by yamadapc** [comment]
tbh; I should probably use use STM, it's included as a dependency for `concurrent-output` now and it'll be fully available on GHC 7.6. Will fix in a sec. I don't know if this causes the library to be actually broken, but let's fix it to avoid trouble.

**Comment by axman6** [comment]
Has this been fixed yet? I'm seeing exactly the behaviour this is supposed to prevent, namely ETA and duration don't work at all for me.

**Comment by yamadapc** [comment]
I believed it had been fixed. STM is a dependency now; I need to port
things to use it, instead of this little hack.

**Comment by axman6** [comment]
Maybe the Stockage version needs updating then, I had to clone this repo to get proper elapsed and ETA times

**Comment by yamadapc** [comment]
Fixed in 0.3.3.0; will open another issue to port to STM
